### ISSUE TITLE
ReadIndex readers pool has reached its max

### ISSUE THREAD

**Issue by TheAngryByrd** [EventStore/EventStore#1528]
I'm seeing errors in the log for one of my eventstore instances

```
Error during processing ReadAllEventsForward request.
Object pool 'ReadIndex readers pool' has reached its max limit for items: 14
```

Is this an error I should be worried about, is there something I should be setting?

**Comment by hayley-jean** [comment]
Hi @TheAngryByrd 

Are you seeing any other errors in the logs?
How much load is your Event Store instance under?
How many simultaneous reads are being run against your Event Store at a time?

**Comment by MadKat13** [comment]
We have the exact same error. Especially seen when scavenging. Currently we have 150 chunks, which will be scavenged.
On our productive system there was an error which terminated the service and the scavenge could never finish. On test however it got also an exception, but just skipped the chunk.

Our CPU is at 40% and at peak (while scavenging) we have somewhere around 80MB/s reads. 

Here with Exception and restart:
![image](https://user-images.githubusercontent.com/7972860/46086036-9945fc80-c1a7-11e8-935d-aa2b830dc65c.png)


Here with Exception and continue:
![image](https://user-images.githubusercontent.com/7972860/46086450-66503880-c1a8-11e8-94a0-d93e0db00729.png)

**Comment by shaan1337** [comment]
Fixed with #1737 
The issue occured when --reader-threads-count is increased from 4 to a larger value.
